pkg/platform/packet/packet.go

```diff
@@ -614,6 +616,25 @@ func (c *config) validateOSVersion() hcl.Diagnostics {
 	return diagnostics
 }
 
+func (c *config) checkCPUManagerPolicy() hcl.Diagnostics {
+	var diagnostics hcl.Diagnostics
+
+	for _, w := range c.WorkerPools {
+		switch w.CPUManagerPolicy {
+		case "", "none", "static":
+			continue
+		default:
+			diagnostics = append(diagnostics, &hcl.Diagnostic{
+				Severity: hcl.DiagError,
+				Summary:  "invalid cpu_manager_policy",
+				Detail:   fmt.Sprintf("Worker pool '%s' has invalid cpu_manager_policy '%s'", w.Name, w.CPUManagerPolicy),
+			})
+		}
+	}
+
+	return diagnostics
+}
```

Comment:
Can we make this type and validation rule generic so there is no duplicated code and tests?